lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/PoisonTest.cs

```diff
@@ -87,45 +88,48 @@ public void OnNext(IDriverStarted value)
 
             public void OnNext(IAllocatedEvaluator value)
```

Comment:
We have two separate needs here: one, to test Poison functionality itself (if we inject failure, failure happens), two, test various scenarios (events received and each event having proper data associated with it) depending on where failure is injected and when it happens. Second need is REEF-1304 which I'm working on. So we don't need to have exhaustive coverage of scenarios in this test.

To keep functionality of this test the same, we can extend delay of poisoning, so that task has enough time to be created.